src/app/components/experiments/create-device-output/create-device-output.component.ts

```diff
@@ -24,77 +24,60 @@ export class CreateDeviceOutputComponent implements OnInit {
   deviceOutputForm: FormGroup;
   submitted: boolean;
   timestamp: number;
+  experimentId: string;
   static PATH: any = 'device-outputs/create';
 
   constructor(private deviceOutputService: DeviceOutputService, private deviceService: DeviceService,
     private route: ActivatedRoute, private formBuilder: FormBuilder, private deviceExperimentService: DeviceExperimentService,
     private router: Router, private outputTypeService: OutputTypeService, private tooltipService: TooltipService,
-    private snackBar: MatSnackBar) {
+    private snackBar: MatSnackBar, private experimentService: ExperimentService) { }
+
+  ngOnInit() {
+    this.experimentId = this.experimentService.experimentId;
+    this.onExperimentUpdate();
+    this.experimentService.$experimentId.subscribe((experimentId) => {
+      this.experimentId = experimentId;
+      this.onExperimentUpdate();
```

Comment:
why does `onExperimentUpdate` get called twice in the init?

Reply:
It's called when the component is loaded and then it's called when there are updates to the current experiment id